### ISSUE TITLE
Command google returned: Command 'google' is disabled: None

### ISSUE THREAD

**Issue by MCharles001** [Significant-Gravitas/Auto-GPT#4210]
### ⚠️ Search for existing issues first ⚠️

- [X] I have searched the existing issues, and there is no existing issue for my problem

### Which Operating System are you using?

Windows

### Which version of Auto-GPT are you using?

Latest Release

### GPT-3 or GPT-4?

GPT-3.5

### Steps to reproduce 🕹

- run run.bat
- load your AI
- Ask for the AI to use google command (because it isn't using it for some reason)
- Watch as the AI uses the google command and returns: SYSTEM:  Command google returned: Command 'google' is disabled: None

### Current behavior 😯

It should be using the google command to search up, but it is not, unless i tell him to do it doesn't do it and when i tell him to do it and he does it, this output pops: SYSTEM:  Command google returned: Command 'google' is disabled: None.

Yes i have my API keys on the .env file, it has nothing to do with that. It has access to the entire web.

### Expected behavior 🤔

It would use the google command with a query and get a search result, and continue to get to its goal

### Your prompt 📝

```yaml
ai_goals:
- Utilize reliable sources to gather the latest population data for all countries
  in the world.
- Organize the data in alphabetical order by country name.
- Save the data in a file named Population.txt.
- Ensure that the data is accurate and up-to-date.
- Provide a summary report of any errors or inconsistencies encountered during the
  data gathering process.
ai_name: PopGPT
ai_role: an autonomous agent that specializes in gathering and organizing population
  data for all countries in the world, and saving them in a file named Population.txt
  in the correct order.
api_budget: 4.52

```

### Your Logs 📒

```log
2023-05-14 22:11:21,661 INFO GPT3.5 Only Mode:  ENABLED
2023-05-14 22:11:22,072 INFO NEWS: Welcome to Auto-GPT!
2023-05-14 22:11:22,073 INFO NEWS: Below you'll find the latest Auto-GPT News and updates regarding features!
2023-05-14 22:11:22,073 INFO NEWS: If you don't wish to see this message, you can run Auto-GPT with the --skip-news flag.
2023-05-14 22:11:22,073 INFO NEWS: 
2023-05-14 22:11:22,073 INFO NEWS: Website and Documentation Site 📰📖
2023-05-14 22:11:22,074 INFO NEWS: Check out https://agpt.co, the official news & updates site for Auto-GPT!
2023-05-14 22:11:22,074 INFO NEWS: The documentation also has a place here, at https://docs.agpt.co
2023-05-14 22:11:22,074 INFO NEWS: 
2023-05-14 22:11:22,074 INFO NEWS: 🚀 v0.3.0 Release 🚀
2023-05-14 22:11:22,074 INFO NEWS: Over a week and 275 pull requests have passed since v0.2.2, and we are happy to announce
2023-05-14 22:11:22,074 INFO NEWS: the release of v0.3.0! From now on, we will be focusing on major improvements rather
2023-05-14 22:11:22,075 INFO NEWS: than bugfixes, as we feel stability has reached a reasonable level. Most remaining
2023-05-14 22:11:22,075 INFO NEWS: issues relate to limitations in prompt generation and the memory system, which will be
2023-05-14 22:11:22,075 INFO NEWS: the focus of our efforts for the next release.
2023-05-14 22:11:22,075 INFO NEWS: 
2023-05-14 22:11:22,075 INFO NEWS: Highlights and notable changes in this release:
2023-05-14 22:11:22,076 INFO NEWS: 
2023-05-14 22:11:22,076 INFO NEWS: Plugin support 🔌
2023-05-14 22:11:22,076 INFO NEWS: Auto-GPT now has support for plugins! With plugins, you can extend Auto-GPT's abilities,
2023-05-14 22:11:22,076 INFO NEWS: adding support for third-party services and more.
2023-05-14 22:11:22,076 INFO NEWS: See https://github.com/Significant-Gravitas/Auto-GPT-Plugins for instructions and available plugins.
2023-05-14 22:11:22,076 INFO NEWS: 
2023-05-14 22:11:22,076 INFO NEWS: Changes to Docker configuration 🐋
2023-05-14 22:11:22,077 INFO NEWS: The workdir has been changed from /home/appuser to /app.
2023-05-14 22:11:22,077 INFO NEWS: Be sure to update any volume mounts accordingly!
2023-05-14 22:11:22,077 INFO NEWS: 
2023-05-14 22:11:22,077 INFO NEWS: ⚠️ Command `send_tweet` is DEPRECATED, and will be removed in v0.4.0 ⚠️
2023-05-14 22:11:22,077 INFO NEWS: Twitter functionality (and more) is now covered by plugins, see [Plugin support 🔌]
2023-05-14 22:11:22,077 INFO NEWS: 
2023-05-14 22:11:22,357 INFO Welcome back!  Would you like me to return to being PopGPT?
2023-05-14 22:11:22,610 INFO  Asking user via keyboard...
2023-05-14 22:21:30,913 INFO PopGPT has been created with the following details:
2023-05-14 22:21:31,173 INFO Name: PopGPT
2023-05-14 22:21:31,233 INFO Role: an autonomous agent that specializes in gathering and organizing population data for all countries in the world, and saving them in a file named Population.txt in the correct order.
2023-05-14 22:21:31,671 INFO Goals: 
2023-05-14 22:21:31,746 INFO - Utilize reliable sources to gather the latest population data for all countries in the world.
2023-05-14 22:21:32,159 INFO - Organize the data in alphabetical order by country name.
2023-05-14 22:21:32,420 INFO - Save the data in a file named Population.txt.
2023-05-14 22:21:32,708 INFO - Ensure that the data is accurate and up-to-date.
2023-05-14 22:21:32,946 INFO - Provide a summary report of any errors or inconsistencies encountered during the data gathering process.
2023-05-14 22:21:33,322 INFO Using memory of type: LocalCache
2023-05-14 22:21:33,484 INFO Using Browser: chrome
2023-05-14 22:21:51,095 INFO  THOUGHTS: I will start by browsing the internet to find reliable sources for population data. I will then write a Python script to scrape the data and save it to a file. Finally, I will organize the data in alphabetical order by country name and save it to a file named Population.txt. I will also need to ensure that the data is accurate and up-to-date.
2023-05-14 22:21:51,701 INFO REASONING: Browsing the internet is the best way to find reliable sources for population data. Writing a Python script to scrape the data will allow me to automate the process and save time. Organizing the data in alphabetical order by country name will make it easier to read and analyze. Saving the data to a file named Population.txt will make it easy to access and share. Ensuring that the data is accurate and up-to-date is important to maintain the integrity of the data.
2023-05-14 22:21:52,361 INFO PLAN: 
2023-05-14 22:21:52,441 INFO -  Browse the internet to find reliable sources for population data
2023-05-14 22:21:52,771 INFO -  Write a Python script to scrape the data and save it to a file
2023-05-14 22:21:53,068 INFO -  Organize the data in alphabetical order by country name and save it to a file named Population.txt
2023-05-14 22:21:53,444 INFO -  Ensure that the data is accurate and up-to-date
2023-05-14 22:21:53,704 INFO CRITICISM: I need to ensure that the sources I use for population data are reliable and up-to-date. I also need to make sure that my Python script is efficient and accurate in scraping the data. Additionally, I need to double-check the data to ensure that it is organized correctly and that there are no errors or inconsistencies.
2023-05-14 22:21:54,304 INFO NEXT ACTION:  COMMAND = browse_website  ARGUMENTS = {'url': 'https://www.worldometers.info/world-population/population-by-country/', 'question': 'reliable sources for population data'}
2023-05-14 22:21:54,646 INFO  Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands'n' to exit program, or enter feedback for ...
2023-05-14 22:21:54,646 INFO  Asking user via keyboard...
2023-05-14 22:22:29,664 INFO SYSTEM:  Human feedback: Use: COMMAND = google ARGUMENTS = {'query': ''}
2023-05-14 22:22:44,082 INFO  THOUGHTS: I will use the Google search engine to find reliable sources for population data. I will search for 'reliable sources for population data' and review the results to find the most trustworthy sources. Once I have identified the sources, I will use them to gather the population data for all countries in the world.
2023-05-14 22:22:44,668 INFO REASONING: Using the Google search engine will allow me to quickly find a variety of sources for population data. Searching for 'reliable sources for population data' will help me find sources that are trustworthy and accurate. Once I have identified the sources, I can use them to gather the population data for all countries in the world.
2023-05-14 22:22:45,259 INFO PLAN: 
2023-05-14 22:22:45,348 INFO -  Use the Google search engine to find reliable sources for population data
2023-05-14 22:22:45,690 INFO -  Search for 'reliable sources for population data' and review the results to find the most trustworthy sources
2023-05-14 22:22:46,031 INFO -  Use the identified sources to gather the population data for all countries in the world
2023-05-14 22:22:46,415 INFO CRITICISM: I need to ensure that the sources I use for population data are reliable and up-to-date. I also need to be careful to avoid sources that may be biased or inaccurate. Additionally, I need to make sure that I am gathering data for all countries in the world and not just a subset.
2023-05-14 22:22:47,043 INFO NEXT ACTION:  COMMAND = google  ARGUMENTS = {'query': 'reliable sources for population data'}
2023-05-14 22:22:47,359 INFO  Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands'n' to exit program, or enter feedback for ...
2023-05-14 22:22:47,359 INFO  Asking user via keyboard...
2023-05-14 22:22:50,345 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-05-14 22:22:50,518 INFO SYSTEM:  Command google returned: Command 'google' is disabled: None
2023-05-14 22:23:05,836 INFO  THOUGHTS: I will use the DuckDuckGo search engine to find reliable sources for population data. I will search for 'reliable sources for population data' and review the results to find the most trustworthy sources. Once I have identified the sources, I will use them to gather the population data for all countries in the world.
2023-05-14 22:23:06,437 INFO REASONING: Using the DuckDuckGo search engine will allow me to quickly find a variety of sources for population data. Searching for 'reliable sources for population data' will help me find sources that are trustworthy and accurate. Once I have identified the sources, I can use them to gather the population data for all countries in the world.
2023-05-14 22:23:07,039 INFO PLAN: 
2023-05-14 22:23:07,113 INFO -  Use the DuckDuckGo search engine to find reliable sources for population data
2023-05-14 22:23:07,413 INFO -  Search for 'reliable sources for population data' and review the results to find the most trustworthy sources
2023-05-14 22:23:07,789 INFO -  Use the identified sources to gather the population data for all countries in the world
2023-05-14 22:23:08,116 INFO CRITICISM: I need to ensure that the sources I use for population data are reliable and up-to-date. I also need to be careful to avoid sources that may be biased or inaccurate. Additionally, I need to make sure that I am gathering data for all countries in the world and not just a subset.
2023-05-14 22:23:08,736 INFO NEXT ACTION:  COMMAND = browse_website  ARGUMENTS = {'url': 'https://duckduckgo.com/', 'question': 'reliable sources for population data'}
2023-05-14 22:23:09,052 INFO  Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands'n' to exit program, or enter feedback for ...
2023-05-14 22:23:09,052 INFO  Asking user via keyboard...

```

**Comment by pimlinders** [comment]
+1. I'm experiencing the exact same issue, i'm using GPT 4.

**Comment by kinance** [comment]
@MCharles001 "Yes i have my API keys on the .env file" - do you mean you have put the google search API key and the custom search engine ID?

**Comment by kinance** [comment]
can you try amending the google command to google_search in google_search.py, e.g.?
![image](https://github.com/Significant-Gravitas/Auto-GPT/assets/6927568/87ff8fe3-9cc4-47cd-91a0-f83b65b99c5d)

**Comment by MCharles001** [comment]
> @MCharles001 "Yes i have my API keys on the .env file" - do you mean you have put the google search API key and the custom search engine ID?

Yes i've done that.

> can you try amending the google command to google_search in google_search.py, e.g.? ![image](https://user-images.githubusercontent.com/6927568/239582766-87ff8fe3-9cc4-47cd-91a0-f83b65b99c5d.png) if this works, there's a bug where the 2 commands are having same name while 1 of them is supposed to be disable.

Ok this, actually made the AI get to use the google_search command by himself but it threw an error:
```
 THOUGHTS:  I need to gather the latest population data for all countries in the world and save it in a file named Population.txt in alphabetical order. I will start by searching for reliable sources of population data.
REASONING:  I need to find reliable sources of population data to ensure that the data I gather is accurate and up-to-date.
PLAN:
-  Search for reliable sources of population data
-  Gather the latest population data for all countries in the world
-  Organize the data in alphabetical order by country name
-  Save the data in a file named Population.txt
CRITICISM:  I need to ensure that the sources I use are reliable and that the data I gather is accurate and up-to-date.
NEXT ACTION:  COMMAND = google_search ARGUMENTS = {'query': 'reliable sources of population data'}
  Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands, 'n' to exit program, or enter feedback for ...
  Asking user via keyboard...
Input:y
-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-=
SYSTEM:  Command google_search returned: Error: Object of type bytes is not JSON serializable
 THOUGHTS:  I need to find reliable sources of population data. I will search for government websites that provide population data.
REASONING:  Government websites are likely to provide reliable and up-to-date population data.
PLAN:
-  Search for government websites that provide population data
-  Gather the latest population data for all countries in the world
-  Organize the data in alphabetical order by country name
-  Save the data in a file named Population.txt
CRITICISM:  I need to ensure that the sources I use are reliable and that the data I gather is accurate and up-to-date.
NEXT ACTION:  COMMAND = google_search ARGUMENTS = {'query': 'government websites that provide population data'}
  Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands, 'n' to exit program, or enter feedback for ...
  Asking user via keyboard...
Input:y
-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-=
SYSTEM:  Command google_search returned: Error: Object of type bytes is not JSON serializable
more here: https://docs.agpt.co/setup/#getting-an-api-key
 THOUGHTS:  I need to find reliable sources of population data. I will search for a list of government websites that provide population data.
REASONING:  A list of government websites that provide population data is likely to be more comprehensive and reliable than a single website.
PLAN:
-  Search for a list of government websites that provide population data
-  Gather the latest population data for all countries in the world
-  Organize the data in alphabetical order by country name
-  Save the data in a file named Population.txt
CRITICISM:  I need to ensure that the sources I use are reliable and that the data I gather is accurate and up-to-date.
NEXT ACTION:  COMMAND = google_search ARGUMENTS = {'query': 'list of government websites that provide population data'}
  Enter 'y' to authorise command, 'y -N' to run N continuous commands, 's' to run self-feedback commands, 'n' to exit program, or enter feedback for ...
  Asking user via keyboard...
Input:

```
It seems like the error is "Object of type bytes is not JSON serializable" I don't have an idea of what this means.

**Comment by MCharles001** [comment]
> Also, the custom search engine ID is correctly set?

What do you mean by this? if you mean if the ID of the CSE is on .env, Yes it is.

**Comment by kinance** [comment]
#1769 is improving the error logging for better troubleshooting.

**Comment by FrancisVarga** [comment]
SYSTEM:  Command google returned: []

Ill have that same error too...

**Comment by aleksei-liap** [comment]
I have the same issue . CSE is configured correct, tested it separately in browser.

**Comment by jerzydziewierz** [comment]
+1 the same issue, google search seems broken.

**Comment by aleksei-liap** [comment]
I'm not sure how is if for you Guys, but for me , it's not a real work around, since it's just just `browse_website `command instead of `google`. Anybody made `google` command on M chip ?

**Comment by MCharles001** [comment]
**This issue was fixed on Release v0.4.0**

I tested it on docker by myself with GPT 3.5, and it works perfectly fine
